File: USACO/2017-2018 Competition/December Bronze/Problem 2/shuffle.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    // File variables - r/w
    ifstream fin ("shuffle.in");
    ofstream fout ("shuffle.out");

    // Read data
    int N, a, b, c;
    fin >> N;
    int movements[N], cows[N];
    for (int i = 0; i < N; i++) {
        fin >> movements[i];
    }
    for (int i = 0; i < N; i++) {
        fin >> cows[i];
    }

    // Alg
    for (int i = 0; i < 3; i++) {
        int temp[N];
        for (int p = 0; p < N; p++) {
            temp[p] = cows[movements[p]-1];
        }
        for (int p = 0; p < N; p++) {
            cows[p] = temp[p];
        }
    }

    for (int i = 0; i < N;) {
        fout << cows[i++] << "\n";
    }
    
    return 0;
}

File: USACO/Competitions/2017-2018/December Bronze/Problem 2/shuffle.py
N, cows, movements = 0, [], []
with open("shuffle.in", "r") as f:
    N = int(f.readline())
    movements = [int(i) for i in f.readline().split(" ")]
    cows = [int(i) for i in f.readline().split(" ")]

for i in range(3):
    temp = [0]*N
    for p in range(len(movements)):
        temp[p] = cows[movements[p]-1]
    cows = temp

with open("shuffle.out", "w") as f:
    for i in cows:
        f.write("%s\n" % i)